### ISSUE TITLE
web app

### ISSUE THREAD

**Issue by joshp23** [tomboy-notes/tomboy#63]
I wrote a [php script](https://github.com/joshp23/Gnoted) for accessing tomboy formatted notes. It supports themes and notebooks, has optional password protection, and I hope to include note creation and editing. I'm looking for feedback and possible direction on adding web-based note editing (without the api).

**Comment by davidbannon** [comment]
Well done Josh, another string to the Tomboy bow.
I also believe we need some new (tomboy) front opened up, you may not have noted my post at the tail end of  https://github.com/tomboy-notes/tomboy/issues/54   I am taking an "old school" approach and suggesting a rewrite using Free Pascal and Lazarus. Have a basic editor that reads and writes Tomboy files in less than 1000 line code, needs only one binary on Linux or Windows, no dependencies. Its the dependencies thats killing Tomboy right now.

David

**Comment by alex-ter** [comment]
I'm on my phone now, so can't check this out, but have you looked into integration or possible duplication with Grauphel? In general it would be quite nice to have an easily deployable Tomboy server with online editing and Grauphel is almost there, barring the fact it's not developed for a while and is intended for Own/NextCloud as the "runtime". Having it revived and standalone could be cool. Are you by any chance interested? :)

**Comment by joshp23** [comment]
alex-ter, I am. Just need the time!

**Comment by joshp23** [comment]
incidentally, I found [this](https://github.com/kimsg1984/tomboy_web_service), which seems interesting...

**Comment by joshp23** [comment]
I forked the repo from my previous comment, attempted to clean up the code a bit and do some translations via Google translate. My fork can be found [here](https://github.com/joshp23/tomboy_web_service).

The heavy lifting for the notes is done with python and php, and there is a javascript editor. It is currently written to work with a sync folder from Tomboy, but I would intend to have it work with `~/local/share/tomboy` directly, as my scheme works with that folder and syncs with dropbox/syncthing type syncing. 

Would love some thoughts!

**Comment by davidbannon** [comment]
Josh, I suggest keeping the sync via Tomboy repository model, it 
provides a reasonable interface for different products to connect to. By 
hitting Tomboy's own notes, you are just a bit close I think.

I have Tomboy-NG syncing via file sync repo (so far) flawlessly and am 
cleaning up  few other things before I release it.

David

**Comment by joshp23** [comment]
Just an update. I have been working on integrating quiljs with a delta translator to tomboy note format.

Am close. When I get it hooked into grauphel for nextcloud and gnoted I will turn my attention to grauphel stand alone. 

Shouldn't be to bad for single users... However multiuser might be an issue. Also grauphel works with flat file storage... And using sql would ve better... So it might take some time for that.

Huzzah!

**Comment by joshp23** [comment]
Web editing a bit closer to reality with a Quilljs delta [renderer](https://github.com/joshp23/denote.js) for Tomboy Notes format.

**Comment by davidbannon** [comment]
Nice work Josh.
My next release of tomboy-ng wll be the last-ish feature release for now and I have been thinking, superficially, how to make notes available to mobile. And thinking a read only webapp might be stage one.
I'm thinking a server that I can connect to from the desktop app, squirt a copy of current notes up there and then view those notes as a Webapp.....
I have a suitable (hardware) server but do wonder if other potential users would have that ready access ?
Davo

**Comment by joshp23** [comment]
If a suitable script could be written to manage multiple users it might be worth while to seek out a hosting co. willing to sponsor such a project with some space. I'll see what I can manage in terms of user management. Perhaps oauth2...

 Grauphel standalone is on my radar as well.

So... One might use the gnoted script as is on a remote server or a home server... Or a more robust iteration of the script might be made to manage multiple users and a generouse sponsor or developer could offer it out.

I might be willing, under thr rihht conditions, to do such a thing and just leave an open donation ask on the interface or some such thing.

Of course... Setting up a drop-box sync with Tomdroid is an old go-to that just works.

How is implimenting tomboy sync api coming?

**Comment by davidbannon** [comment]
My own feelings are few users these days would want their notes, unencrypted, on some hosting company's server. OwnCloud or just some server software running on a home PC would be fine for sync but pretty useless for a Webapp. 

Encrypting the notes is an obvious solution but where ?  File comparability with Tomboy is a good thing but maybe its a built in "Tomboy Import and Export" capability ? I would like to consider moving the notes into a database and maybe thats when encryption happens ?

I have standard Tomboy file sync (ie to Google Drive or DropBox) running fine but no network version. The file sync inherits from a generic sync class, the intention being only transport things need change to implement a network Oauth version (eg pointed to rainy) but I must admit it (the protocol) turned out a lot more complicated than I expected - not all that confident I want to tackle the network  :-)

**Comment by joshp23** [comment]
Indeed. My initial thoughts for Gnoted were for personal use on home server or personally owned vps. Implimenting multi-user still makes sense for use with family, etc.

Grauphel-nextcloud syncs to databse, and I was already thinking about looking into encryption there after getting editing/creation locked down a but more. I agree it's critical.

I'm not familiar with Pascal at all... But network sync is pretty critical for me, else I would have just stuck with Gnote.
